docs/artefacts/offramps.md

```diff
@@ -638,3 +638,114 @@ offramp:
       compress_logs: true
       region: europe
 ```
+
+### KV
+
+The kv offramp is intended to allow for a decoupled way of persisting and retrieving state in a non blocking way.
+
+Events sent to the KV offramp are commands.
+
+Example:
+```yaml
+  - id: kv
+    type: kv
+    linked: true     # this needs to be true
+    codec: json
+    config:
+      dir: "temp/kv" # directory to store data in
+```
+
+The following commands are supported:
+
+
+#### get
+
+Fetches the data for a given key.
+
+**Request**:
+```json
+{"get": {"key": "<string|binary>"}}
+```
+
+**Response**:
+```json
+{"ok": "<decoded>"} // key was found, the format of decoded depends on the codec (does NOT have to be a string)
+null,               // key was not found
+```
+
+#### put
+
+Writes a value to a key, returns the old value if there was any.
+
+**Request**:
+```json
+{"put": {
+  "key": "<string|binary>",
+  "value": "<to encode>" // the format of value depends on the codec (does NOT have to be a string)
```

Comment:
this comment is highlighted in bad red, as it is illegal inside a json document. You might just put it below the json code-sample.

Reply:
Please use `js` for highlighting. Treating this as javascript covers json and comments :)

Reply:
`jsonc` also seems to work (at least on github)